actors/builtin/miner/miner_actor.go

```diff
@@ -1273,12 +1273,15 @@ func processSkippedFaults(rt Runtime, st *State, store adt.Store, currDeadline *
 
 	// Penalize new skipped faults and retracted recoveries
 	// This differs from declaring faults "on time", where retracting recoveries doesn't attract an extra penalty
-	penalizeFaultSectors := append(newFaultSectors, retractedRecoveryInfos...)
-	penalty, err := unlockUndeclaredFaultPenalty(st, store, minerInfo.SectorSize, currEpoch, epochReward, currentTotalPower, penalizeFaultSectors)
+	faultPenalty, err := unlockUndeclaredFaultPenalty(st, store, minerInfo.SectorSize, currEpoch, epochReward, currentTotalPower, newFaultSectors)
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to charge fault fee")
+
+	// retracted recoveries are penalized at a higher rate
+	retractedPenalty, err := unlockUndeclaredLateFaultPenalty(st, store, minerInfo.SectorSize, currEpoch, epochReward, currentTotalPower, retractedRecoveryInfos)
```

Comment:
I think there's a misunderstanding here. If declared "skipped" in a PoSt submission, both new faults and retracted recoveries should have the same SP penalty. The 2*SP penalty applies if no PoSt submission is made for a sector and it was either a) not declared faulty, or b) declared recovered. But this skipped faults code path shouldn't change.

Reply:
got it.